crates/core/src/node/network_bridge/p2p_protoc.rs

```diff
@@ -75,6 +75,40 @@ impl P2pBridge {
             log_register: Arc::new(event_register),
         }
     }
+
+    /// Send Unsubscribed messages to upstream peers after subscription tree pruning.
+    ///
+    /// This is called after `prune_connection()` or `remove_client_from_all_subscriptions()`
+    /// to notify upstream peers that we're no longer subscribed to certain contracts.
+    pub(crate) async fn send_prune_notifications(
+        &self,
+        notifications: Vec<(freenet_stdlib::prelude::ContractKey, PeerKeyLocation)>,
+    ) {
+        if notifications.is_empty() {
+            return;
+        }
+
+        let own_location = self.op_manager.ring.connection_manager.own_location();
+
+        for (contract_key, upstream) in notifications {
```

Comment:
is possible to do this concurrently and not awaiting for each send sequentially?